app-crypt/certbot/certbot-9999.ebuild

```diff
@@ -146,40 +146,73 @@ RDEPEND="
 # 	)
 # "
 
+# Note: "docs" is not an actual directory under "S", they are actually
+# under each modules, see python_compile_all redefinition, but keep
+# this instruction enabled for dependency configuration.
+distutils_enable_sphinx docs \
+	dev-python/sphinx-rtd-theme
 distutils_enable_tests pytest
 
-certbot_dirs=()
+CERTBOT_DIRS=()
+# Stores temporary modules docs in each subdirectories,
+# will be used for HTML_DOCS
+CERTBOT_DOCS="${T}/docs"
 
-my_certbot_dirs_listing() {
+src_prepare() {
+	default
+
+	# set CERTBOT_DIRS
 	local base module
 	for base in "${CERTBOT_BASE[@]}"; do
-		certbot_dirs+=("${base}")
+		CERTBOT_DIRS+=("${base}")
 	done
 	for module in "${CERTBOT_MODULES_EXTRA[@]}"; do
 		use "certbot-${module}" \
-			&& certbot_dirs+=("certbot-${module}")
+			&& CERTBOT_DIRS+=("certbot-${module}")
 	done
-}
-
-src_prepare() {
-	default
 
-	my_certbot_dirs_listing
+	# Used to build documentation
+	mkdir "${CERTBOT_DOCS}" || die
 }
 
 python_compile() {
 	local dir
-	for dir in "${certbot_dirs[@]}"; do
+	for dir in "${CERTBOT_DIRS[@]}"; do
 		pushd "${dir}" > /dev/null || die
 
 		distutils-r1_python_compile
-		# Delete previous build directory to avoid collision.
-		rm -rf "${BUILD_DIR}/build"
 
 		popd > /dev/null || die
 	done
 }
 
+# Used to build documentation
+python_compile_all() {
+	use doc || return
+
+	local dir
+	for dir in "${CERTBOT_DIRS[@]}"; do
+		# There is no documentation in certbot-apache or certbot-nginx.
+		if [[ "${dir}" = "certbot-apache" ]] || [[ "${dir}" = "certbot-nginx" ]]; then
```

Comment:
```suggestion
		if has "${dir}" "certbot-apache" "certbot-nginx"; then
```

Reply:
Never heard off, is this Portage and/or eclass specific?

Reply:
It's in PMS.